### ISSUE TITLE
PHPMD baseline is ouput in "random" order : please output <violation> tags sorted by filepath then rule.

### ISSUE THREAD

**Issue by LLyaudet** [phpmd/phpmd#1308]
- PHPMD version: 2.15.0
- PHP Version: 8.5.4
- Installation type: (phar file/ composer) composer
- Operating System / Distribution & Version: (Windows 10 / CentOS 7.6 / ...) Ubuntu 26.04

### Current Behavior

I add XML comments in baseline to justify that some error is unchecked.
But when I generate a new baseline,
unfortunately errors are not always exactly in the same order
and it messes up diff.

### Expected Behavior

```xml
  <violation rule="PHPMD\Rule\Controversial\SomeRule"
    file="SomeFilepath"
  />
```
Sort by lexicographic order on filepath first,
then break ties by sorting on rule.

### Steps To Reproduce:

I maintained my repository with scripts that generate the new baseline at each build (and keep it in temp folder) and diff it to the old baseline.
If there is a difference, I do a manual update afterward.

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by AJenbo** [comment]
file path then line number seems to be the intended order and also makes sens when one method violates multiple rules for the same reason (to long and to complex)

**Comment by LLyaudet** [comment]
Hello @AJenbo :),
thanks for your fast answer.
I didn't knew it was line number, but note that baseline disable some rule most of the time by a per-file basis.
Optionally, a method attribute is given in the XML.
It makes sense, but for my use case, it is important that the diff is minimized.
I found a solution by myself.
I used XSLT to reorder the XML before diffing:
<https://github.com/LLyaudet/DevOrSysAdminScripts/blob/main/build_and_checks_dependencies/phpmd_baseline.xslt>
<https://github.com/LLyaudet/DevOrSysAdminScripts/blob/61f933c5170d0fdafe6b7f5127a0be4c34d548fa/build_and_checks_dependencies/common_build_and_checks.exec.sh#L713-L732>
It seems to work equally well after being reordered.
Maybe I could suggest adding a feature option to phpmd:
`--baseline-attributes-order=`
with possible values generated from:
- `file`
- `first-line`
- `rule`
- `method`

Thus the default would be:
`--baseline-attributes-order=file,first-line,rule`

What I use would be:
`--baseline-attributes-order=file,rule`

etc.

Best regards,
     Laurent Lyaudet
